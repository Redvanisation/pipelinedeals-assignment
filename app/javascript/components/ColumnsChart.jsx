import React from 'react';
import { Bar, defaults } from 'react-chartjs-2';
import PropTypes from 'prop-types';

defaults.global.legend.display = false;
defaults.global.defaultFontFamily = "'PT Sans', sans-serif"


const ColumnsChart = ({ isLoading, chartData }) => {


  return (
    <div>
      <h2 className="main-title">Column chart displaying total $ deal value in each stage</h2>
      {
        !isLoading
          ? (
            <Bar
              className="bar"
              data={chartData}
              options={{}}
           />
          )
          : <h2 className="loading">Content is loading!</h2>
      }
      
    </div>
  );
};

ColumnsChart.defaultProps = {
  isLoading: true
}

ColumnsChart.propTypes = {
  isLoading: PropTypes.bool,
  chartData: PropTypes.instanceOf(Object).isRequired
}

export default ColumnsChart;
