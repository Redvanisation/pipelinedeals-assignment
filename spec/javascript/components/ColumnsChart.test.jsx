import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
// import { expect } from 'chai';
import { Bar } from 'react-chartjs-2';


import ColumnsChart from '../../../app/javascript/components/ColumnsChart';

describe('ColumnsChart component', () => {
  let wrapper;
  let isLoading = true;
  
  let props = {
    isLoading: isLoading, 
    chartData: {           
      labels: [],
      datasets: [
      {
        data: [],
      },
      ]}
    };

  const flushPromise = () => new Promise(resolve => setImmediate(resolve));


  beforeEach(() => { wrapper = shallow(<ColumnsChart {...props} />) });
  
  it('renders correctly', () => {
    expect(wrapper).not.toBeNull();
  });

  it('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('props are passed correctly', () => {
    expect(wrapper.props()).toBeTruthy();
  });

  it('renders a title', () => {
    expect(wrapper.find('.main-title')).toHaveLength(1);
    expect(wrapper.find('.main-title').text().includes('Column chart displaying')).toBe(true);
  });

  it('renders a loading text before getting the data', () => {
    expect(wrapper.find('.loading')).toHaveLength(1);
    expect(wrapper.find('.loading').text().includes('Content is loading!')).toBe(true);
  });

  it('renders the Bar chart after getting the data', async () => {
    await flushPromise();
    wrapper.update();
    wrapper.setProps({ isLoading: false });
    expect(wrapper.find(Bar)).toHaveLength(1);
  })
});