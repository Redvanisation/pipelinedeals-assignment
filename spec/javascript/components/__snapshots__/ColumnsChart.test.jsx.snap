// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnsChart component Matches snapshot 1`] = `ShallowWrapper {}`;
